unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/BackpackEditorHUDController.cs

```diff
@@ -82,10 +87,56 @@ public class BackpackEditorHUDController
             view.OnColorChanged += OnWearableColorChanged;
             view.OnColorPickerToggle += OnColorPickerToggled;
             view.OnAvatarUpdated += OnAvatarUpdated;
+            view.OnOutfitsOpened += OnOutfitsOpened;
+
+            outfitsController.OnOutfitEquipped += OnOutfitEquipped;
 
             SetVisibility(dataStore.HUDs.avatarEditorVisible.Get(), saveAvatar: false);
         }
 
+        private void OnOutfitEquipped(OutfitItem outfit)
+        {
+            Dictionary<string, WearableItem> keyValuePairs = new Dictionary<string, WearableItem>(model.wearables);
+            foreach (KeyValuePair<string, WearableItem> keyValuePair in keyValuePairs)
+                UnEquipWearable(keyValuePair.Key, UnequipWearableSource.None, false, false);
+
+            foreach (string forcedCategory in model.forceRender)
+                UpdateOverrideHides(forcedCategory, false, false);
+
+            outfitLoadCancellationToken = new CancellationTokenSource();
+            LoadAndEquipOutfitWearables(outfit, outfitLoadCancellationToken.Token).Forget();
+        }
+
+        private async UniTaskVoid LoadAndEquipOutfitWearables(OutfitItem outfit, CancellationToken cancellationToken)
+        {
+            if (!wearablesCatalogService.WearablesCatalog.ContainsKey(outfit.outfit.bodyShape))
+                await wearablesCatalogService.RequestWearableAsync(outfit.outfit.bodyShape, cancellationToken);
+            foreach (string outfitWearable in outfit.outfit.wearables)
+            {
+                if (!wearablesCatalogService.WearablesCatalog.ContainsKey(outfitWearable))
+                    await wearablesCatalogService.RequestWearableAsync(outfitWearable, cancellationToken);
+            }
+
+            if(!string.IsNullOrEmpty(outfit.outfit.bodyShape))
```

Comment:
if the body shape is empty shouldnt all the operation fail?

Reply:
True, it was a leftover of some tests I was doing